packages/realm_dart/test/realm_logger_test.dart

```diff
@@ -1,211 +1,105 @@
+    Isolate.run(() {
+      Realm.logger.log(RealmLogLevel.trace, 'Hey');
     });
-
-    expect(actual, LoggedMessage(RealmLogLevel.detail, "123"));
   });
 
-  test('Realm.logger supports logging from multiple isolates', () async {
-    List<LoggedMessage> actual = await Isolate.run(() async {
-      var completer = Completer<void>();
-      final result = <LoggedMessage>[];
-
-      Realm.logger.level = RealmLogLevel.all;
-      Realm.logger.onRecord.listen((event) {
-        if (event.message == "stop") {
-          return completer.complete();
-        }
-
-        result.add(LoggedMessage(event.level, event.message));
-      });
-
-      // run a second isolate to listen logging specific log level messages
-      await Isolate.run(() async {
-        var completer2 = Completer<void>();
-        Realm.logger.level = RealmLogLevel.error;
-        Realm.logger.onRecord.listen((event) {
-          if ((event.level == RealmLogLevel.error && event.message != "2") || (event.level == RealmLogLevel.trace && event.message != "3")) {
-            throw RealmError("Unexpected message ${LoggedMessage(event.level, event.message)}");
-          }
-
-          completer2.complete();
-        });
+  test('Trace in root isolate seen in subisolate', () async {
+    Realm.logger.setLogLevel(RealmLogLevel.off);
+    Realm.logger.setLogLevel(RealmLogLevel.all, category: RealmLogCategory.realm.sdk);
 
-        Future<void> logTestMessage(Level level, String message) async {
-          completer2 = Completer<void>();
-          RealmInternal.logMessageForTesting(level, message);
-          if (Realm.logger.level == RealmLogLevel.off) {
-            Future<void>.delayed(const Duration(milliseconds: 10)).then((value) {
-              completer2.complete();
-            }).ignore();
-          }
-          await completer2.future;
-        }
-
-        await logTestMessage(RealmLogLevel.error, "2");
-        await logTestMessage(RealmLogLevel.error, "2");
-
-        // turn off second isolate
-        Realm.logger.level = RealmLogLevel.off;
-
-        // log another message. second isoalte should not get it
-        await logTestMessage(RealmLogLevel.error, "first only");
+    final trace = Isolate.run(() async {
+      return (await Realm.logger.onRecord.first).message;
+    });
+    await Future<void>.delayed(const Duration(milliseconds: 100)); // yield
+    expectLater(trace, completion('Hey'));
+    Realm.logger.log(RealmLogLevel.trace, 'Hey');
+  });
 
-        // turn on second isolate
-        Realm.logger.level = RealmLogLevel.trace;
+  test('RealmLogger hookup logging', () async {
+    Realm.logger.setLogLevel(RealmLogLevel.off);
+    Realm.logger.setLogLevel(RealmLogLevel.all, category: RealmLogCategory.realm.sdk);
 
-        // log a another message
-        await logTestMessage(RealmLogLevel.trace, "3");
+    final logger = Logger.detached('Test');
+    Realm.logger.onRecord.forEach((r) => logger.log(r.level.level, r.message));
```

Comment:
Do we need to clean this up?

Reply:
?

Reply:
Sorry, was late and wasn't super clear. I meant that when you do `Realm.logger.onRecord.listen` you get a subscription you can then cancel. With `forEach` you don't seem to get that, so wouldn't that code continue executing even after this particular test completes.

Reply:
😬 eagle eye..

Reply:
Fixed